test/vault_helpers.go

```diff
@@ -115,7 +111,10 @@ func deployCluster(t *testing.T, amiId string, examplesDir string, uniqueId stri
 		},
 	}
 	test_structure.SaveTerraformOptions(t, examplesDir, terraformOptions)
-	terraform.InitAndApply(t, terraformOptions)
+	retry.DoWithRetry(t, "Running terraform Init", 3, 10*time.Second, func() (string, error) {
+		return terraform.InitE(t, terraformOptions)
```

Comment:
`terraform init` was sometimes failing to install aws provider because it just fails to make the http request and download the package. I discussed with Josh and he suggested this type of change becomes built-in from terratest, but we should see how it behaves here first.

Reply:
BTW, +1 on building this into Terratest (perhaps with a flag to optionally disable it), as this is definitely a common source of (transient) errors.